spec/apis/event-listeners.spec.js

```diff
@@ -120,39 +120,20 @@ describe(`event listeners after single-spa is started`, () => {
     }
   });
 
-  it(`Does trigger a reroute when the URL changes and urlRerouteOnly is set to false`, async () => {
-    let activeWhenCalls = 0,
-      popstateCalls = 0;
-    const activeWhen = () => activeWhenCalls++;
-    const popstateListener = () => popstateCalls++;
-    const app = { async bootstrap() {}, async mount() {}, async unmount() {} };
-
-    window.addEventListener("popstate", popstateListener);
-    singleSpa.registerApplication("urlRerouteOnly test", app, activeWhen);
-
-    await singleSpa.triggerAppChange();
-
-    const numPopstatesBefore = popstateCalls;
-    const numActiveWhensBefore = activeWhenCalls;
-    history.replaceState({ some: "state" }, document.title);
-    // calling triggerAppChange forcibly increments the counters which is weird for this test
-    // but it also ensures we wait for the reroute to finish (if it's taking place)
+  it(`Fires artificial popstate events with correct target`, async () => {
```

Comment:
I think you can do `aysnc (done)` here if that makes thing easier.

Reply:
`async (done)` is being removed, according to https://jestjs.io/blog/2021/05/25/jest-27#features-coming-with-breaking-changes

Reply:
Oh, good to know